src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -686,6 +687,23 @@ export class BundleDataClient {
       }
     });
 
+    // Verify that a fill sent to an EVM chain has a 20 byte address. If the fill does not, then attempt
+    // to repay the `msg.sender` of the relay transaction. Otherwise, add it to invalid fills.
+    const verifyFill = async (fill: FillWithBlock, spokePoolClient: SpokePoolClient) => {
```

Comment:
We should also verify here that `repaymentChainId` is valid - i.e. is it a currently supported chain?

Reply:
Could also move down here https://github.com/across-protocol/sdk/blob/master/src/clients/BundleDataClient/BundleDataClient.ts#L1282

Reply:
Alternatively, we should just make sure to call getRefundInformationFromFill in this function to load the repayment chain rather than just reading it directly from the event

Reply:
now that I'm looking at the code again, I'm thinking we might want to put this `verifyFill` logic inside `getRefundInformationFromFill`. This is because anytime we call `getRefundInformationFromFill`, for example in `getApproximateRefundsForBlockRange` we also would want to know if the repayment address is being swapped